api/src/main/java/org/pmiops/workbench/api/CohortsController.java

```diff
@@ -135,6 +140,12 @@ public Cohort apply(org.pmiops.workbench.db.model.Cohort cohort) {
       String cohortId, Cohort cohort) {
```

Comment:
Consider moving this into a new updateCohort() method on a new CohortService class. That will allow us to use this logic if updating a cohort from someplace else.

We could potentially refactor some the logic for dealing with Etags and versions and updates into a common helper method, like:

static T <T implements HasVersion> saveWithVersion(String etag, T dbObj, CrudRepository<T, ?> dao) {
  if (!Strings.isNullorEmpty(etag) {
    ...
  }
  try {
    dao.save(dbObj);
  } catch () {
   ...
  }
  dbObj.setVersion(dbVersion.getVersion() + 1);
  return dbObj;
}

Reply:
Suppose we add a command line tool for modifying users or workspaces (or whatever.) We'd want to apply the same logic to that tool.

Or if we wind up adding another endpoint that winds up modifying workspaces or users, that would need this logic as well.

Not sure if we're going to do any of that. But in case we do, I've been recommending that any core business logic we want to enforce about entities like this should go in a service layer separate from the controllers. That way we won't duplicate logic or forget to apply this logic later.